### PR TITLE
Create header

### PR DESCRIPTION
**Pull request by cit280superepicness**
Header added to user_home and mostly complete (still need to get Spring Security working so that users can only see "Login" and "Register" on the header when they are looged out and "Logout" when they are logged in. 

Footer complete.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,6 +57,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-jdbc</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
         <!-- end webjar -->
```
Comment (jrmullen) [inline pom.xml]: remove comment

**Comment by jrmullen** [comment]
All images were added to the pull request. Please choose only the images we will be using.

File: pom.xml [hunk 2]
```diff
@@ -57,6 +57,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-jdbc</artifactId>
         </dependency>
+        <dependency>
```
Comment (jrmullen) [inline pom.xml]: Move security spring dependency outside of the webjar comments

**Comment by jrmullen** [comment]
I see you say that spring security is not configured. Is this going to cause problems? Have you tested that all pages are functional?

File: src/main/resources/static/index.html [hunk 3]
```diff
@@ -1,134 +0,0 @@
-<!DOCTYPE html>
```
Comment (jrmullen) [inline src/main/resources/static/index.html:1]: Is this file deleted?

File: pom.xml (hunk 2 above)
Comment (cit280superepicness) [inline pom.xml]: Fixed this

File: src/main/resources/static/index.html (hunk 3 above)
Comment (cit280superepicness) [reply src/main/resources/static/index.html:1]: No, just empty index.html was empty in my branch for whatever reason. I fixed this. Sorry, I will be more careful from now on with things like this. I was doing a ton of things to try to get the login and register popups working with my header last night and must have committed this by mistake and then not reverted this file when I reverted the rest of my changes.  Sorry about that.

**Comment by cit280superepicness** [comment]
Fixed this.

**Comment by cit280superepicness** [comment]
What I meant by that is that I'm not sure if I have Spring Security 100%
working as intended. If I did not have th:if="${#authorization.
expression('!isAuthenticated()')} commented out in header.html, it would
evaluate to NULL and cause an error. However, it is commented out, so this
is not a problem.

The database errors I am working to resolve (due to my local database not
being configured correctly yet) when I register or login keep me from
viewing the result.html page to ensure it is functional. However, I can say
that the rest of the pages are functional.

**Comment by jrmullen** [comment]
result.html is not associated with database in any way. I don't know what version of the code you have, but if it is the newest version result.html does not exist. It was removed when db was implemented. I only created it to verify that my form was working as intended since I created it before the DB was set up.

So - you either have a DB issue, or a result.html data issue. The two are independent of one another. Check and see which, because if it is a result.html issue then there is a bigger problem then just not having the db configured correctly.

**Comment by cit280superepicness** [comment]
Sorry about that. I really need to check over my work a lot more thoroughly
to make sure I don't have any extra files and that I grab the latest remote
master before I create a pull request next time. I did a "git fetch"  and
then "git rebase origin master" to get the latest changes. result.html is
no longer there and I have the latest files now.

I will try to get everything in order regarding the header and user_home,
so we can show it in our demo in class Tuesday.

Renee

**Comment by cit280superepicness** [comment]
Good news! I got my database set up and register working (it inserts data
into the User table) and I have no result.html anymore.

The bad news is that I am getting a NullPointerException at line 42 of
User.java when I try to login with any user from the database. If this is
not a bug for you as well, I will see where I went wrong (again!) to ensure
that I have the correct files.

 @AssertTrue()

public boolean isDifferentPass() {
    return !passwordConfirm.equals(password) ? false : true;
}

**Comment by jrmullen** [comment]
The register page is the only page that I set up using DB.

I didn't set up the login page because I thought spring security would take care of that. Is this correct? Anyways the login page is not functional. It will throw errors etc.

As for line 42 it may be a bug and we may have to reconsider. I am using the above code snippet in the register page to confirm that the password and confirm password fields are matching. It may have some weird side effects we will have to look into it further.

**Comment by jrmullen** [comment]
Also note that you do not need to most recent code to create a pull request. You just have to know what is functional with what code you have.

**Comment by cit280superepicness** [comment]
When you say that you didn't set up the register page, what do you mean
exactly? Did you mean login instead of register?

Spring security should be used to redirect to the Watchdog index.html if
the user is not logged in. This can probably be achieved by storing
credentials and checking that they are valid before loading a page.

**Comment by jrmullen** [comment]
login. I made an edit but edits don't show in emails. Sorry.

**Comment by cit280superepicness** [comment]
It's no problem. With login, I think we will need it connected to the
database to ensure the user has an accouny. If there is another you have in
mind, please let me know.
I imagine the process would something like make sure the user's credentials
match what is in the database, basically, select \* from user where
user_email and user_password = what the user entered for email and
password. If there is a match in the database, then we will probably set
something in spring security to say that the user is authorized, so they
can access pages like log viewer, user home , etc. Then save the logged in
user's data in an instance of the User class for the duration that they are
logged in. We may use a session object of some type got when the user is
logged in  like in PHP but I'm not sure.

**Comment by jrmullen** [comment]
The branch is unable to be merged. There are conflicts in the pom.xml folder. Can you please resolve these?

**Comment by cit280superepicness** [comment]
Thanks for the info. I'll take a look. I believe it is because I added a
couple dependencies for Spring Security.

**Comment by jrmullen** [comment]
IntelliJ should have an to show and resolve conflicts. Google it

**Comment by cit280superepicness** [comment]
Still looking into resolving it. I can view the differences in the files in
the Git shell and on IntelliJ and click append changes. However, there is
no indication that clicking append changes anything or if it appends one
line or a few lines of text that are different. The only changes between
the pom.xml files I added a couple lines for the Spring security, added a
comment, and moved the dependency for the Spring JDBC out of the webjars
part of the file.

**Comment by cit280superepicness** [comment]
Nm, I think I found it.

**Comment by cit280superepicness** [comment]
I threw out my pom.xml file and merged with master. I wanted to just append
my changes but unfortunately ran out of time to get it working before I
have to go to bed.

You may have to add the following to pom.xml in origin/master for Spring
Security to work.

```
    <dependency>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-security</artifactId>
    </dependency>
```

**Comment by cit280superepicness** [comment]
Sorry! I messed up master. I am now trying to revert it. I really don't
know what I'm doing. My apologies about this.

**Comment by cit280superepicness** [comment]
Reverted origin/master. I do not want to mess things up further here.
Please just don't include my pom.xml when you merge create_header with
master and it should merge without issues. Then after you merge, to get the
application to compile without errors from Spring Security, please add the
following to origin/master's pom.xml.

   <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-security</artifactId>
        </dependency>

**Pull request merged**